src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -111,6 +114,11 @@ static <T> Blob constructBlobAndMetadata(
           compressedChunkData = serializedChunk.chunkData.toByteArray();
           chunkLength = compressedChunkData.length;
           compressedChunkDataSize = chunkLength;
+
+          if (internalParameterProvider.setIcebergSpecificFieldsInEp()) {
```

Comment:
this calculation can happen wholly inside the pre-existing if(setIcebergSpecificFieldsInEp() section at line 151?

Reply:
Moved to line 151.